pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -174,6 +174,7 @@ type Controller struct {
 func (e *Controller) Run(ctx context.Context, workers int) {
 	defer utilruntime.HandleCrash()
 	defer e.queue.ShutDown()
+	defer e.eventBroadcaster.Shutdown()
```

Comment:
what should be the right order?
first shutdown the queue and the eventBroadcaster last?

Reply:
Yeah - it should be the opposite - we should first shutdown the queue and only the broadcaster
[currently it's the opposite]